api/sheet.go

```diff
@@ -46,6 +46,8 @@ type Sheet struct {
 	Instance   *Instance `jsonapi:"relation,instance"`
 	DatabaseID *int      `jsonapi:"attr,databaseId"`
 	Database   *Database `jsonapi:"relation,database"`
+	ProjectID  *int      `jsonapi:"attr,projectId"`
```

Comment:
qq: is projectID (and/or databaseID) is required? If so, the type *int should not be a pointer here, but int instead. Same below.

Reply:
The `databaseID` is optional because of the connection doesn't need to specify a database in some cases. So I think `projectID` is optional too. Confirm again @tianzhou

Reply:
BTW, if the `projectID` is required, must change the logic of creating tabs.

when creating the sheet, must set the `projectId`, otherwise, the tabs can not be created as expected.

Reply:
ProjectID is required

Reply:
We have a default project which is project ID 1. When we sync the instance schema, the synced databases will be put under that default project. We do this because database must belong to a project as well.

The sheet here has a similar situation when we create a new tab. So an easy fix is also set it to the default project.

On the other hand, if the current user already has a project context (e.g. user switches to a different project in the left navigation bar), then it's better to set the project ID to that project.